optimum/onnxruntime/base.py

```diff
@@ -178,21 +179,28 @@ def __init__(
                 self.value_sequence_length_idx = -1
 
     def prepare_inputs_for_merged(
-        self, input_ids: Optional[torch.LongTensor] = None, past_key_values: Optional[List[torch.FloatTensor]] = None
+        self,
+        input_ids: Union[None, torch.LongTensor, np.ndarray] = None,
+        past_key_values: Optional[Union[Tuple[torch.FloatTensor], Tuple[np.ndarray]]] = None,
+        use_torch: bool = False,
     ):
+        constructor = torch if use_torch is True else np
         if past_key_values is None and self.parent_model.use_merged:
             # Uses "no past" branch of a merged decoder
-            use_cache_branch = torch.full((1,), False).to(self.device)
+            use_cache_branch = constructor.full((1,), False)
         elif past_key_values is not None and self.parent_model.use_merged:
             # Uses "with past" branch of a merged decoder
-            use_cache_branch = torch.full((1,), True).to(self.device)
+            use_cache_branch = constructor.full((1,), True)
```

Comment:
Can we merge this and make the second argument equal to `past_key_values is not None`?

Reply:
yes good catch!